src/main/kotlin/org/elm/workspace/compiler/ElmBackAction.kt

```diff
@@ -0,0 +1,29 @@
+package org.elm.workspace.compiler
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.util.messages.Topic
+import org.elm.ide.toolwindow.ElmCompilerPanel
+
+class ElmBackAction(private val elmCompilerPanel: ElmCompilerPanel) : AnAction() {
```

Comment:
I'm no expert on the IntelliJ action system, but I'm pretty sure that they are supposed to be "beans" (no-arg constructor).

I took a brief look at the implementors of the `AnAction` abstract class, and, as far as I can tell, everything uses the  `DataContext` system for acquiring information from other parts of the system.

Last fall I made some notes on how `DataContext` and company work. Here's a gist: https://gist.github.com/klazuka/23a9894253fb6878f95ab74ba1b75bd9

So rather than taking the `ElmCompilerPanel` in the constructor, I think you should instead be obtaining it from the `DataContext` (which also probably means that on the `ElmCompilerPanel` side you will need to somehow "register" it).

Reply:
Felt weird doing this anyway 😉